itest/assets_test.go

```diff
@@ -1363,6 +1391,22 @@ func createAssetHodlInvoice(t *testing.T, dstRfqPeer, dst *HarnessNode,
 	}
 }
 
+// addGroupModeOpt may add a group key option to the opts array, if the group
+// mode boolean is true.
+func addGroupModeOpt(opts *[]payOpt, groupMode bool, groupID []byte) {
+	if groupMode {
+		*opts = append(*opts, withGroupKey(groupID))
```

Comment:
How about just a new functional option:
```
func withMaybeGroupKey(groupMode bool, groupKey []byte) payOpt {
	return func(c *payConfig) {
		if !groupMode {
			return
		}

		c.groupKey = groupKey
	}
}
```

And the same for `withInvGroupKey`? Should reduce the diff quite a bit below.